Homework3/mostFrequentElement.c

```diff
@@ -0,0 +1,148 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+
+void swap(int *first, int *second) {
+    int temp = *first;
+    *first = *second;
+    *second = temp;
+}
+
+void insertionSort(int low, int high, int array[]) {
+    for (int i = low; i <= high; ++i) {
+        int j = i;
+        while (j > low && array[j - 1] > array[j]) {
+            swap(&array[j - 1] , &array[j]);
+            --j;
+        }
+    }
+}
+
+int semiQsort(int low, int high, int array[]) {
+    int middle = (low + high) / 2;
+
+    int *tempArray = (int*)(calloc(3, sizeof(int)));
+
+    tempArray[0] = array[low];
+    tempArray[1] = array[middle];
+    tempArray[2] = array[high];
+    insertionSort(0, 2, tempArray);
+
+    array[0] = tempArray[0];
+    array[middle] = tempArray[1];
+    array[high] = tempArray[2];
+
+    free(tempArray);
+
+    swap(&array[1], &array[middle]);
+    int border = high - 1;
+    const int firstNumber = array[low + 1];
+
+    for (int i = low + 2; i < border; ++i) {
+        if (array[i] >= firstNumber) {
+            while (array[border] >= firstNumber && i < border) {
+                --border;
+            }
+            swap(&array[i], &array[border]);
+            --border;
+        }
+    }
+
+    if (firstNumber < array[border]) {
+        --border;
+    }
+
+    if (!(low == high)) {
+        swap(&array[low + 1], &array[border]);;
+    }
+
+    return border;
+}
+
+void qsortRecursion(int low, int high, int array[]) {
+    if (low < high) {
+        if (high - low < 9) {
+            insertionSort(low, high, array);
+        } else {
+            int border = semiQsort(low, high, array);
+            qsortRecursion(low, border - 1, array);
+            qsortRecursion(border + 1, high, array);
+        }
+    }
+}
+
+int mostFrequentNumber(int array[], int arrayLength) {
+    unsigned int maxRow = 0;
+    unsigned int recentCount = 1;
+    int maxRowElement = 0;
+    
+    //создание копии массива, так как функция не всегда должна менять проверяемый массив
```

Comment:
```suggestion
    // создание копии массива, так как функция не всегда должна менять проверяемый массив
```
Символ комментария всегда отделяется от текста пробелом. Но вообще, очень правильно